app/src/main/java/org/kiwix/kiwixmobile/intro/IntroActivity.kt

```diff
@@ -0,0 +1,129 @@
+import androidx.viewpager.widget.ViewPager
+import androidx.viewpager.widget.ViewPager.OnPageChangeListener
+import kotlinx.android.synthetic.main.activity_intro.*
+import org.kiwix.kiwixmobile.R
+import org.kiwix.kiwixmobile.core.Intents.internal
+import org.kiwix.kiwixmobile.core.base.BaseActivity
+import org.kiwix.kiwixmobile.core.di.components.CoreComponent
+import org.kiwix.kiwixmobile.core.main.CoreMainActivity
+import org.kiwix.kiwixmobile.kiwixActivityComponent
+import java.util.Timer
+import java.util.TimerTask
+import javax.inject.Inject
+
+class IntroActivity : BaseActivity(), IntroContract.View {
+
+  companion object {
+    private const val timerDelay: Long = 0
+    private const val timerPeriod: Long = 2000
+    private const val animationDuration: Long = 800
+  }
+
+  private val handler = Handler()
+  private val timer = Timer()
+
+  @Inject
+  internal lateinit var presenter: IntroContract.Presenter
+  private lateinit var airPlane: ImageView
+  private var currentPage = 0
+  private lateinit var views: Array<View>
+
+  @Suppress("EmptyFunctionBlock")
+  private val pageChangeListener: OnPageChangeListener = object : OnPageChangeListener {
+    override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+    }
+
+    override fun onPageSelected(position: Int) {
+      if (position == 1) {
+        airPlane.visibility = View.VISIBLE
+        airPlane.animate().translationX(airPlane.width.toFloat()).duration = animationDuration
+      } else {
+        airPlane.visibility = View.INVISIBLE
+        airPlane.animate().translationX(-airPlane.width.toFloat())
+      }
+      currentPage = position
+    }
+
+    override fun onPageScrollStateChanged(state: Int) {
+      if (state == ViewPager.SCROLL_STATE_DRAGGING) {
+        dismissAutoRotate()
+      }
+    }
+  }
+
+  override fun injection(coreComponent: CoreComponent) {
+    this.kiwixActivityComponent.inject(this)
+  }
+
+  override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_intro)
+    get_started.setOnClickListener { startMainActivity() }
+    views = arrayOf(
+      layoutInflater.inflate(R.layout.item_intro_1, view_pager, false),
+      layoutInflater.inflate(R.layout.item_intro_2, view_pager, false)
+    )
+    val introPagerAdapter = IntroPagerAdapter(views)
+    view_pager.adapter = introPagerAdapter
+    tab_indicator.setViewPager(view_pager)
+    airPlane = views[1].findViewById(R.id.airplane)
+    view_pager.addOnPageChangeListener(pageChangeListener)
+    timer.schedule(object : TimerTask() {
+      override fun run() {
+        handler.post {
+          if (currentPage == views.size) {
+            currentPage = 0
+          }
+          view_pager.setCurrentItem(currentPage++, true)
+        }
+      }
+    }, timerDelay, timerPeriod)
+    views.forEach {
+      it.findViewById<View>(R.id.root).setOnClickListener { dismissAutoRotate() }
```

Comment:
actually we don't need to call findViewById at all here because these views are already the root containers so

```
it.setOnClickListener{}
```

is identical and in fact a bit faster

Reply:
Done👍